### ISSUE TITLE
get title by load file

### ISSUE THREAD

**Issue by YaBrady** [PHPOffice/PHPWord#1180]
Hello
I want to judge the type between text and title by IOFactory::load($fileName)，But their type are all text.How can i distinguish them?

**Comment by FBnil** [comment]
You have to look at the style name, one is "normal", the other "HeadingX" (exact naming depends on how you created the titles). 

`<w:p><w:pPr><w:pStyle w:val="Heading2"/>`.text is here..`</w:p>`

use `get_class_methods($phpWord)` and you will see getters that you can call and use. (i.e.`->getStyle()`) then dump the variable and see the properties.

**Comment by YaBrady** [comment]
@FBnil
 Which type of object 

code--------------
$fileName = "./xxx.docx";
$phpword = IOFactory::load($fileName);
$section = $phpword->getSections();

echo '<pre>';
//This is a title
 $obj=$section[5]->getElements()[12];
--------------------
the class of $obj is Text, it only has getParagraphStyle() and getFontStyle(), I can't find "HeadingX" in two return content, sad..

**Comment by FBnil** [comment]
Hm... all text is paragraph, and some paragraphs are titles. This grabs the first line and displays the Paragraph Style name (which, granted, could be changed, but is often named as title) 
```
	$testDoc = storage_path('Untitled.docx');
	$phpWord = \PhpOffice\PhpWord\IOFactory::load($testDoc, 'Word2007');
	$to = $phpWord->getSections()[0]->getElements()[0]->getTextObject();
	dd([
		'pstyle' => $to->getParagraphStyle()->getStyleName(), 
		'text' => $to->getText()
	]);
```
This is a somewhat working thing, sorry for the mess:
```
function ExtractData($obj, $nested = 0){
	$txt = "";
	if(method_exists($obj, 'getSections')) {
		foreach ($obj->getSections() as $section) {
			$txt .= "!" . ExtractData($section, $nested+1);
		}
	}else if (method_exists($obj, 'getElements')) {
		foreach ($obj->getElements() as $element) {
			$txt .= " " . ExtractData($element, $nested+1);
		}
	}else if (method_exists($obj, 'getText')) {
		$paragraphStyle = '?';
		if (method_exists($obj, 'getTextObject')) {
				$tobj = $obj->getTextObject();
				$paragraphStyle = $tobj->getParagraphStyle()->getStyleName();
		}
		if (get_class($obj) == "PhpOffice\PhpWord\Element\ListItem" && $obj->getDepth()<51) {
			$ndx = $obj->getElementIndex();
			$ns = $obj->getStyle()->getNumStyle();
			$ni = $obj->getStyle()->getNumId();
			#dd(['getText'=>$obj,  'get_class_methods'=>get_class_methods($obj),'getStyle'=>$obj->getStyle(),'styleMethods'=>get_class_methods($obj->getStyle())]);
			$txt .= " $ndx:".$obj->getText() ."($ni:$ns)($paragraphStyle)";
		}else{
			$txt .= "|";
		}
	}
	return $txt;
}


$echo_this_variable = ExtractData($phpWord);
```

**Comment by FBnil** [comment]
And don't be sad, it sounds like depressive sad...

**Comment by YaBrady** [comment]
@FBnil Thx! I find the titile
Don't worry ,I am a positive guy.

It will get the object by getElements()[0], it does not have the method of getTextObject(), Am i phpword's version too low?  :)

Anyway, with your help i find the solution
```
$to = $phpWord->getSections()[0]->getElements()[0];
dd([
	'pstyle' => $to->getParagraphStyle()->getStyleName(), 
	'text' => $to->getText()
]);
```
when  $to is **TextRun** and **the  content  is  normal text**, it will return
```
array:2 [▼
  "pstyle" =>null
  "text" => null
]
```
when $to is heading 1, it will return 
```
array:2 [▼
  "pstyle" => "2"
  "text" => null
]
```
Similarly, heading 2 is {pstyle"=>"3"}, heading 3 is {pstyle"=>"4"}

when the $to is Text, it just change  parameter "text" to  it's content
Ex:
array:2 [▼
  "pstyle" => "4"
  "text" => "4.1.6 This is the heading 3"
]

**Comment by kobbe0** [comment]
I got an related issue.
All Headlines in my docx also show as Text/TextRun.
I was able to get the Style name Rubrik2 (Heading2 in English).
I can see all properties of this Paragraph style using:
$paragraph_style = $section_element->getParagraphStyle();

But I am not able to figure out what makes this Text a Headline or what Depth it has.

Also not able to find out how to get "all styles available", is it not possible?

Any advice? :)
